src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T & string): GroupsMap<T> {
+  const groups: GroupsMap<T> = {};
+
+  items.forEach((item: T) => {
+    const groupKey: T [keyof T & string] = item[key];
```

Comment:
`keyof T` - it's a type of certain key, but `item[key]` - it's not a key, it's a value
```suggestion
    const groupKey: string = item[key];
```